### PR TITLE
executor: don't warn about unused `TARGETARCH`,`TARGETOS`,`TARGETPLATFORM`

### PR DESCRIPTION
**Pull request by flouthoc**
Buildah warns about TARETARCH, TARGETOS, TARGETPLATFORM in a multi-arch build however its not very useful for end-users and there are cases where this can be left untouched so do not warn about test default build args

Closes: https://github.com/containers/buildah/issues/4485

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note
executor: don't warn about unused TARGETARCH,TARGETOS,TARGETPLATFORM
```

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@nalind @vrothberg @giuseppe @umohnani8 @TomSweeneyRedHat PTAL

File: tests/bud.bats [hunk 1]
```diff
@@ -610,6 +610,20 @@ _EOF
...
 }
 
+# Verify: https://github.com/containers/buildah/issues/4485
+@test "No default warning for TARGETARCH, TARGETOS, TARGETPLATFORM " {
+  local contextdir=${TEST_SCRATCH_DIR}/bud/platform
+  mkdir -p $contextdir
+
+  cat > $contextdir/Dockerfile << _EOF
+FROM alpine
+_EOF
```
Comment (nalind) [inline tests/bud.bats:620]: This file doesn't need to be generated every time this test is run.

File: tests/bud.bats [hunk 2]
```diff
@@ -610,6 +610,20 @@ _EOF
...
+
+  run_buildah build $WITH_POLICY_JSON --platform=linux/amd64,linux/arm64 -f $contextdir/Dockerfile
```
Comment (nalind) [inline tests/bud.bats:622]: https://github.com/containers/buildah/issues/4319 is a similar issue.  Might as well cover that case as well.

**Comment by TomSweeneyRedHat** [comment]
@flouthoc all kinds of test unhappiness on this one.  Other than @nalind 's comments. LGTM

**Pull request merged**